### DISCUSSION TITLE
Alarm Logging Service Improvements

### DISCUSSION THREAD

**Discussion by tanviash** [ControlSystemStudio/phoebus#1802] [General]
Advance search dialog for alarm logging application
Put a REST api in front of the alarm logging backend

**Comment by shroffk** [comment]
For improving the advance search ui we could try to standardize of something similar to the logtable search UI as shown below

![image](https://user-images.githubusercontent.com/2111304/110336788-36512480-7ff3-11eb-9837-716d1f6b9a77.png)

Have some prepopulated drop down menus for fixed option search 
Use the same from and end time search

**Reply by shroffk** [reply]
@tanviash do you have an ideas for a nicer way of visualizing the "<" and ">" expand/collapse button

**Comment by kasemir** [comment]
Not sure ..

The 'alarm logger' for now just sends the messages to elastic, and you can then use any of the tools in the elastic ecosystem to look for timeline, top alarms, most noisy alarm trigger PV etc.

Elastic has a 'REST' API, and there are already several tools like kibana.

Do you have time to implement another kibana, or a custom REST interface for which we'd have to create all the clients ourselves ? 


![Screen Shot 2021-03-08 at 9 47 48 AM](https://user-images.githubusercontent.com/1932421/110336983-68fb1d00-7ff3-11eb-90e6-81114b4ce30d.png)
![Screen Shot 2021-03-08 at 9 47 55 AM](https://user-images.githubusercontent.com/1932421/110336990-6993b380-7ff3-11eb-9b32-c21db2fec81d.png)
![Screen Shot 2021-03-08 at 9 48 02 AM](https://user-images.githubusercontent.com/1932421/110336992-6a2c4a00-7ff3-11eb-9893-30fe8a346d30.png)

**Comment by shroffk** [comment]
The elastic REST api remains in place.

The REST api for the logging service is a lower priority project... but it would help if we don't want to rewrite the client each time there is an upgrade in the olog logging elastic backend version.

So logging REST api should not affect the use of Kibana and other clients.
